server/src/plugins/groups-decorator.ts

```diff
@@ -0,0 +1,46 @@
+import fp from 'fastify-plugin';
+import { FastifyReply, FastifyInstance } from 'fastify';
+import { IncomingMessage, ServerResponse, Server } from 'http';
+import { UserRole, auth0USer } from '../types';
+
+export class jwtIncomingMessage extends IncomingMessage {
+  user?: {
+    'https://poap.xyz/roles': [ string ];
+    iss: string;
+    sub: string;
+    aud: [ string, ];
+    iat: Date;
+    exp: Date;
+    azp: string;
+    scope: string;
+  }
+}
+
+export function getUserRoles(user: auth0USer | any){
```

Comment:
Please let's change `auth0USer` to `Auth0User`

Reply:
If possible or for future references prefer `unknown` type over `any`. [Rational](https://mariusschulz.com/blog/the-unknown-type-in-typescript)

Reply:
but if i use `unknown` in the next line y can't use it as an object, this line `return user['https://poap.xyz/roles']` does not compile, any suggestion?

Reply:
Yes that's was my concern that's why I was refering if was possible or for future parts were we need `any`.
For debugging TS errors what I recommend is to to use the code that's shown in the console or in VSCode when you hover on the line. I.e:
<img width="751" alt="Captura de Pantalla 2019-12-04 a la(s) 09 19 56" src="https://user-images.githubusercontent.com/2475912/70142146-59b25480-1677-11ea-8616-0ac4812fb345.png">

Then you can go and find in Google `TS7053 typescript` and you will have suggestions on how to fix that error.